src/selectors/operations.ts

```diff
@@ -0,0 +1,22 @@
+import { createSelector } from "reselect";
+import { IRootState } from '../reducers';
+import { IOperationsState, INotification, ITransaction } from "../reducers/operations";
+
+const operations = (state: IRootState) => state.operations;
+
+export type LabeledNotification = {type: 'notification', id: string} & INotification;
+export type LabeledTransaction = {type: 'transaction', id: string} & ITransaction;
+export type LabeledOperation = LabeledNotification | LabeledTransaction;
```

Comment:
i dont quite understand this. an operation is either a transaction or a notification? seems like we have been thinking of an operation as being one or more transactions, with associated notifications (one more more per transaction, pending, success, failure...). so this structure is confusing to me

Reply:
the selector should really not be doing any optimization for rendering, it should just be about filtering the data. i think we need to do some brainstorming about how to better organize/architect operations and notifications, which should be totally separate concepts. they feel too intertwined to me right now

Reply:
one could imagine notifications unrelated to operations, and operations can definitely have several associated notifications...

Reply:
I thought about separating them into modules but figured there's not much benefit in that. But maybe your'e right.

Reply:
separated them in: https://github.com/daostack/alchemy/pull/110/commits/3d3584b77885eade7101d2d126b16ab2b9bdbf83

Reply:
@tibetsprague Re, selector, I thought the role of selectors is to map the canonical redux state into the UI state (i.e. something that's useful for the UI).